src/QsCompiler/TextProcessor/ParsingPrimitives.fs

```diff
@@ -38,6 +38,25 @@ let internal isSymbolStart c = isAsciiLetter c || c = '_'
 /// returns true if the given char is a valid symbol continuation - i.e. if it is an ascii letter, a digit, or an underscore
 let internal isSymbolContinuation c = isAsciiLetter c || isDigit c || c = '_'
 
+/// returns true if the given char is a valid start symbol for an identifier: Underscore or a Unicode character of classes Lu, Ll, Lt, Lm, Lo, Nl.
+/// https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/lexical-structure#identifiers
+let internal isValidIdentifierStart c =
```

Comment:
Taking a closer look, I think it would actually be good to just replace isSymbolStart (in this file) with this implementation.

Reply:
Yes, this should work.